### PR TITLE
WT-7989 Compact quits when running at the same time as a system checkpoint

### PR DESCRIPTION
**Pull request by AlexeyAnisimovAU**
This change adds a csuite test to check that compact and checkpoint can run in parallel. This test is not compatible with windows.

### CODE REVIEW

**Review by etienneptl — APPROVED** [review]
LGTM

File: test/csuite/wt7989_compact_checkpoint/main.c [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+{
+    struct thread_data *td;
+    WT_SESSION *session;
+
+    td = (struct thread_data *)arg;
+
+    testutil_check(td->conn->open_session(td->conn, NULL, NULL, &session));
+
+    if (td->cond != NULL) {
+        /* Make sure checkpoint thread is initialized and waiting for the signal. Sleep for one
```
Comment (hbokhari) [inline test/csuite/wt7989_compact_checkpoint/main.c:188]: Malformed multi-line comment.

**Review by hbokhari — APPROVED** [review]
LGTM with a minor fix.

File: test/csuite/wt7989_compact_checkpoint/main.c (hunk 1 above)
Comment (AlexeyAnisimovAU) [reply test/csuite/wt7989_compact_checkpoint/main.c]: Done.

**Comment by AlexeyAnisimovAU** [comment]
Thank you for the review!

**Pull request merged**